api/genomics/src/main/java/org/pmiops/workbench/genomics/RandomizeVcf.java

```diff
@@ -140,30 +140,49 @@ protected Genotype randomizeGenotype(VariantContext variantContext, Genotype gen
      */
     List<Allele> alleles = new ArrayList<>();
     double genotypeTypeIndex = random.nextDouble();
-    if (genotypeTypeIndex < .0145) {
-      // double no-call
-      return alleles;
-    } else if (genotypeTypeIndex < .8095) {
-      // homref
-      alleles.add(variantContext.getReference());
-      alleles.add(variantContext.getReference());
-    } else if (genotypeTypeIndex < .8654) {
-      // 0/1 het
-      alleles.add(variantContext.getReference());
-      alleles.add(variantContext.getAlternateAllele(0));
-    } else if (genotypeTypeIndex < .9268) {
-      // 1/0 het
-      alleles.add(variantContext.getAlternateAllele(0));
-      alleles.add(variantContext.getReference());
-    } else if (genotypeTypeIndex < .9994
-        || (genotypeTypeIndex >= .9994 && variantContext.getAlternateAlleles().size() == 1)) {
-      // homvar
-      alleles.add(variantContext.getAlternateAllele(0));
-      alleles.add(variantContext.getAlternateAllele(0));
+    if (variantContext.getAlternateAlleles().size() == 1) {
+      if (genotypeTypeIndex < .0145) {
+        // double no-call
+        alleles.add(Allele.NO_CALL);
+        alleles.add(Allele.NO_CALL);
+        return alleles;
+      } else if (genotypeTypeIndex < .8095) {
+        // homref
+        alleles.add(variantContext.getReference());
+        alleles.add(variantContext.getReference());
+      } else if (genotypeTypeIndex < .8654) {
+        // 0/1 het
+        alleles.add(variantContext.getReference());
+        alleles.add(variantContext.getAlternateAllele(0));
+      } else if (genotypeTypeIndex < .9268) {
+        // 1/0 het
+        alleles.add(variantContext.getAlternateAllele(0));
+        alleles.add(variantContext.getReference());
+      } else {
+        // homvar
+        alleles.add(variantContext.getAlternateAllele(0));
+        alleles.add(variantContext.getAlternateAllele(0));
+      }
     } else {
```

Comment:
nit: I think it would read a little quicker if we specified that the size should be equal to 2. I was trying to figure this out earlier while I was reading.

Reply:
that makes sense, I'll flip it around!